File: test/ruby/pairing_param.rb
require_relative "helper"

class TestPairingParam < TestCase
  def test_to_and_init_from_str
    assert_raise RuntimeError do
      PairingParam.init_from_str("type a")
    end
    param = PairingParam.init_from_str(TYPE_A_PARAM)
    assert_equal TYPE_A_PARAM, param.to_str
  end

  def test_gen_type_a
    param = PairingParam.gen_type_a(160, 512)
    assert_match %r{type a}, param.to_str
  end
end

File: test/python/pairing_param.py
from helper import *

class TestPairingParam(unittest.TestCase):
    def test_to_and_init_from_str(self):
        with self.assertRaises(RuntimeError):
            PairingParam.init_from_str("type a")
        param = PairingParam.init_from_str(TYPE_A_PARAM)
        self.assertEqual(param.to_str(), TYPE_A_PARAM)

    def test_gen_type_a(self):
        param = PairingParam.gen_type_a(160, 512)
        self.assertRegex(param.to_str(), "type a")

if __name__ == '__main__':
    unittest.main()
